src/storage/src/source/mysql/replication.rs

```diff
@@ -344,18 +458,19 @@ async fn raw_stream<'a>(
     // which happens to be our `resume_transaction_id`
     // https://dev.mysql.com/doc/refman/8.0/en/replication-options-gtids.html#sysvar_gtid_executed
     // https://dev.mysql.com/doc/dev/mysql-server/latest/classGtid__set.html#ab46da5ceeae0198b90f209b0a8be2a24
-    let seen_gtids = match resume_transaction_id.into() {
-        0 | 1 => {
-            // If we're starting from the beginning of the binlog or the first transaction id
-            // we haven't seen 'anything'
-            Vec::new()
-        }
-        ref a => {
+    let seen_gtids = resume_upper
+        .iter()
+        // If we're starting from the beginning of the binlog or the first transaction id
+        // we haven't seen 'anything'
+        .filter(|partition| Into::<u64>::into(partition.timestamp()) > 1)
```

Comment:
What happens if we pass zero? If MySQL is cool with it we should remove the special case even if transaction zero is never actually observed. If on the other hand transactions are always non-zero we should consider using `NonZeroU64` as the type in `TransactionId`

Reply:
Yes transaction-ids are non-zero. From the docs https://dev.mysql.com/doc/refman/8.0/en/replication-gtids-concepts.html
>  It is not possible for a transaction to have 0 as a sequence number in a GTID. 

I'll try to switch to `NonZeroU64` assuming it doesn't introduce any other complexities

Reply:
Nice. Using `NonZeroU64` will also make the timestamp representation more compact because rust will figure out that it can internally use the value `0` to encode the `Absent` value! https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=01f83b5a3f264cee2176c6bfa50c5c03